src/avram/connection.cr

```diff
@@ -0,0 +1,22 @@
+# Handles the connection to the DB.
+class Avram::Connection
+  def self.open(connection_string : String) : DB::Database
+    conn = self.new(connection_string)
+    conn.try_connection!
+  end
+
+  def initialize(@connection_string : String)
+  end
+
+  def try_connection!
+    begin
+      DB.open(@connection_string)
+    rescue DB::ConnectionRefused
+      raise ConnectionError.new(connection_uri)
+    end
```

Comment:
You can remove the begin and end and it will still work. This is a handy little shortcut for rescuing errors